common/lib/dependabot/file_updaters/vendor_updater.rb

```diff
@@ -27,7 +27,12 @@ def updated_vendor_cache_files(base_directory:)
           )
           changed_paths = status.split("\n").map { |l| l.split(" ") }
           changed_paths.map do |type, path|
+            # The following types are possible to be returned:
+            # M = Modified = Default for DependencyFile
+            # D = Deleted
+            # ?? = Untracked = Created
             deleted = type == "D"
+            created = type == "??"
```

Comment:
This is starting to look like we might need a `type` or `mode` struct/variable that can be either `:modified, :deleted, :created`. Fine to leave this in for now but since these are mutually exclusive it seems like a bit of a smell

Reply:
Nah I think you're right. I just added operation to DependencyFile and removed deleted/created.

Reply:
Ah that's a great change but I'm slightly worried that this api is used externally, I'm going to check out what we need to keep, might be just the `#deleted?` flag. Apologies, I should have mentioned that before

Reply:
Yeah probably only deleted, I'll make sure its still there